skimage/segmentation/chanvese.py

```diff
@@ -0,0 +1,166 @@
+    sizex = U0.shape[1]/10.;
+    sizey = U0.shape[0]/10.;
+    return (np.sin(np.pi/sizey*yv)*(np.sin(np.pi/sizex*xv)))
+
+def _cv_calculate_variation(U0, phi, mu, lambda1, lambda2,
+                            heavyside=_cv_heavyside):
+    """ Calculates the variation of the level set function for
+    one step of the Chan-Vese algorithm. This value is not multipled by
+    the 'virtual time' expressed in the original article.
+    """
+    H = heavyside(phi)
+    Hinv = 1.-H
+    Hsum = abs(np.sum(H))
+    Hinvsum = abs(np.sum(Hinv))
+    c1 = np.sum(U0*H)
+    c2 = np.sum(U0*Hinv)
+    if(Hsum!=0):
+        c1 = c1/Hsum
+    if(Hinvsum!=0):
+        c2 = c2/Hinvsum
+    return _cv_edge_length_term(phi, mu)-lambda1*(U0-c1)**2+lambda2*(U0-c2)**2
+
+def chan_vese(U0, mu, lambda1=1., lambda2=1., tol=1., maxiter=100, dt=1):
+    """ Returns a bool array representing two separate regions
+    separated through the Chan-vese algorithm.
+    
+    Parameters
+    ----------
+    U0 : An 2D array containing floating point values
+        This is the grayscale image that needs to be segmented.
+    mu : A floating point value
+        This is the 'edge length' weight parameter. Higher mu values
+        will produce a 'round' edge, while values closer to zero will
+        detect smaller objects.
+    lambda1 : A floating point value
+        This is the 'difference from average' weight parameter for the
+        output region with value 'True'. If it is lower than lambda2,
+        this region will have a larger range of values than the other.
+    lambda2: A floating point value
+        This is the 'difference from average' weight parameter for the
+        output region with value 'False'. If it is lower than lambda2,
+        this region will have a larger range of values than the other.
+    tol : A positive floating point value
+        This is the 'energy' variation tolerance between iterations.
+        If the difference of 'energy' between two iterations is below
+        this value, the algorithm will assume that the solution was
+        reached.
+    maxiter : An positive integer value
+        This is the maximum number of iterations allowed before the
+        algorithm interrupts itself
+    dt : A floating point value
+        This variable serve as accelerate the algorithm. It is a
+        multiplication factor applied at calculations for each step.
+        While higher values may speed up the algorithm, they may also
+        lead to convergence problems.
+        
+    Returns
+    -------
+    phi : A 2D array of boolean values
+        This is the result of the Chan-Vese algorithm. 'True' and 
+        'False' values have no particular meaning.
+    energies : A list of floating point values
+        This list shows the evolution of the 'energy' for each step of
+        the algorithm. This should allow to check whether the algorithm
+        converged.
+    
+    For an explanation of the Algorithm see:
```

Comment:
See https://github.com/scikit-image/scikit-image/blob/master/skimage/segmentation/_felzenszwalb.py#L40 for example on how to format references in docstring.